compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1223,4 +1223,39 @@ object messages {
            |Attempting to define a field in a method signature after a varargs field is an error.
            |""".stripMargin
   }
+
+  case class AmbiguousImport(name: Names.Name, newPrec: Int, prevPrec: Int, prevCtx: Context)(implicit ctx: Context)
+    extends Message(AmbiguousImportID) {
+
+    import typer.Typer.BindingPrec._
+
+    /** A string which explains how something was bound; Depending on `prec` this is either
+      *      imported by <tree>
+      *  or  defined in <symbol>
+      */
+    private def bindingString(prec: Int, whereFound: Context, qualifier: String = "") =
+      if (isImportPrec(prec)) {
+        ex"""imported$qualifier by ${hl"${whereFound.importInfo}"}"""
+      } else
+        ex"""defined$qualifier in ${hl"${whereFound.owner.toString}"}"""
+
+
+    val msg =
+      s"""|reference to `${hl"$name"}` is ambiguous
+          |it is both ${bindingString(newPrec, ctx)}
+          |and ${bindingString(prevPrec, prevCtx, " subsequently")}""".stripMargin
+
+    val kind = "Reference"
+
+    val explanation =
+      hl"""|The compiler can't decide which of the possible choices you
```

Comment:
I agree, the current formatters don't compose well. @felixmulder and I discussed the need automated line wrapping as well.